tests/src/data/use-cases/db-send-email.spec.ts

```diff
@@ -0,0 +1,68 @@
+import { type EmailService, type LoadGuardianByEmailRepository } from '@/data/protocols'
+import { DbSendEmail } from '@/data/use-cases'
+import { makeFakeEmailService, makeFakeGuardianRepository } from '@/tests/utils'
+
+interface SutTypes {
+  sut: DbSendEmail
+  guardianRepositoryStub: LoadGuardianByEmailRepository
+  emailServiceStub: EmailService
+}
+
+const makeSut = (): SutTypes => {
+  const guardianRepositoryStub = makeFakeGuardianRepository()
+  const emailServiceStub = makeFakeEmailService()
+  const sut = new DbSendEmail({ guardianRepository: guardianRepositoryStub, emailService: emailServiceStub })
+
+  return {
+    sut,
+    guardianRepositoryStub,
+    emailServiceStub
+  }
+}
+
+describe('DbSendEmail', () => {
+  const data = { email: 'any_email' }
+
+  it('Should call GuardianRepository with correct value', async () => {
+    const { sut, guardianRepositoryStub } = makeSut()
+    const loadByEmailSpy = jest.spyOn(guardianRepositoryStub, 'loadByEmail')
+    await sut.send(data)
+    expect(loadByEmailSpy).toHaveBeenCalledWith('any_email')
+  })
+
+  it('Should throw if GuardianRepository throws', async () => {
+    const { sut, guardianRepositoryStub } = makeSut()
+    jest.spyOn(guardianRepositoryStub, 'loadByEmail').mockRejectedValueOnce(new Error())
+    const promise = sut.send(data)
+    await expect(promise).rejects.toThrow()
+  })
+
+  it('Should throw if GuardianRepository returns null', async () => {
```

Comment:
if you use Not found Error class that I mentioned before, you will need to change this test.